pkg/addons2/adminconsole/install.go

```diff
@@ -0,0 +1,194 @@
+package adminconsole
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/embedded-cluster/pkg/addons/registry"
+	"github.com/replicatedhq/embedded-cluster/pkg/helm"
+	"github.com/replicatedhq/embedded-cluster/pkg/kotscli"
+	"github.com/replicatedhq/embedded-cluster/pkg/spinner"
+	"github.com/replicatedhq/embedded-cluster/pkg/versions"
+	"golang.org/x/crypto/bcrypt"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (a *AdminConsole) Install(ctx context.Context, kcli client.Client, writer *spinner.MessageWriter) error {
+	// some resources are not part of the helm chart and need to be created before the chart is installed
+
+	if err := createNamespace(ctx, kcli, namespace); err != nil {
```

Comment:
helm has a --create-namespace flag, can we just set that var in our helm installer to make this unecessary?

Reply:
we need to create some resources before we install the chart, and we need the namespace to exist in order to do that. i've captured part of that in the comment above